File: 61. Rotate List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        int size = 0;
        ListNode gg = head;
        while(gg!=null){
            size++;
            gg = gg.next;
        }
        System.out.println("Size of list - " + size);
        if(k>0 && k>=size && size!=0){
            k = k%size;
        }
        if(head != null && head.next == null){
            return head;
        }
        else if(head != null && head.next.next != null){
            ListNode rr = new ListNode();
            for(int i=0; i<k; i++){
                ListNode curr = head;
                while(curr.next.next != null){
                    curr = curr.next;
                }
                ListNode sl = curr;
                ListNode l = curr.next;
                sl.next = null;
                l.next = head;
                head = l;
            }
            return head;
        }
        else if(head !=null && head.next != null){
            if(k%2 == 0){
                return head;
            }else{
                ListNode kk = head.next;
                head.next.next = head;
                head.next = null;
                return kk;
            }
        }
        else{
            return null;
        }
    }
}

File: 61. Rotate List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if head != None:
            if k==0:
                return head
            length = 1
            curr = head
            last = None
            while curr.next != None:
                length = length +1 
                curr = curr.next
                if curr.next == None:
                    last = curr
            print(length)

            if length == 1:
                return head
            
            k = k%length
            if k==0:
                return head
            newHead = None
            newCurr = head
            for i in range(1, length-k):
                newCurr = newCurr.next
                i = i+1
            print("newCurr - ", newCurr.val)
            newHead = newCurr.next
            last.next = head
            newCurr.next = None

            return newHead
        else:
            return head
        